bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/DiscoveryServiceRegistryOSGITest.groovy

```diff
@@ -440,5 +440,55 @@ class DiscoveryServiceRegistryOSGITest extends OSGiTest {
         assertNotNull extendedDiscoveryServiceMock.discoveryServiceCallback.getExistingDiscoveryResult(thingUID)
     }
 
+    @Test
+    void 'assert that updated discovery result is stored properly in cache'() {
+        ThingUID thingUID = new ThingUID(EXTENDED_BINDING_ID, EXTENDED_THING_TYPE, "foo")
+        
+        
+        AsyncResultWrapper<DiscoveryResult> discoveryListenerResult = new AsyncResultWrapper<DiscoveryResult>()
+        DiscoveryListener discoveryListenerMock = [
+            thingDiscovered: { DiscoveryService source, DiscoveryResult result ->
+                discoveryListenerResult.set result
+
+            },
+            removeOlderThings: { DiscoveryService source, long timestamp, Collection<ThingTypeUID> thingTypeUIDs ->
+                []
+            }
+        ] as DiscoveryListener
+
+        boolean discoveryFinished = false;
+        discoveryServiceRegistry.startScan(new ThingTypeUID(EXTENDED_BINDING_ID, EXTENDED_THING_TYPE), [
+            onFinished: { discoveryFinished = true },
+            onErrorOccurred: { }
+        ] as ScanListener);
+        waitForAssert ({ assertTrue(discoveryFinished) }, 2000 );
```

Comment:
please don't pass any custom timeouts but stick to the defaults here. I know at least to CI servers where this is going to fail occasionally...! 😉 

(same for all the timeouts below)